File: matlab/fred/fred_min.m
% Descargador FRED (fredgraph CSV) — script mínimo
% ----------------------------------------------------------------------------
% Objetivo
%   Descargar el CSV de un gráfico compartido (fredgraph) y guardarlo como CSV tras
%   cargarlo en una tabla de MATLAB.
% Parámetros editables
%   - graphId: identificador del gráfico (parámetro g de la URL)
% Salida
%   - Archivo CSV con la serie del gráfico
% Pasos
%   1) Construir URL fredgraph.csv?g=...
%   2) Leer a tabla y exportar a CSV

% Identificador del gráfico
graphId = '1wmdD';
% Construir la URL
url = sprintf('https://fred.stlouisfed.org/graph/fredgraph.csv?g=%s', graphId);
% Leer la URL directamente a tabla
T = readtable(url, 'FileType', 'text');
% Exportar la tabla a CSV
writetable(T, 'fred_graph_min.csv', 'FileType', 'text');

File: R/fred/fred_min.R
# Descargador mínimo FRED (fredgraph CSV)
#
# Cómo personalizar:
# - Cambie 'graphId' por el identificador del gráfico compartido en FRED
# - Cambie 'out_file' para definir el nombre del archivo de salida (CSV)

library(httr)

# --- Parámetros editables ---
graphId <- "1wmdD"
output_name <- "fred_graph_ejemplo"
out_file <- paste0("data/", output_name, ".csv")

# --- Crear directorio de datos si no existe ---
if (!dir.exists("data")) dir.create("data", recursive = TRUE, showWarnings = FALSE)

# --- Descargar archivo CSV ---
url <- "https://fred.stlouisfed.org/graph/fredgraph.csv"
res <- httr::GET(url, query = list(g = graphId), httr::accept("text/csv"))
if (httr::http_error(res)) stop(sprintf("FRED fredgraph request failed [%s]", httr::status_code(res)))
writeBin(httr::content(res, as = "raw"), out_file)


